canvasapi/util.py

```diff
@@ -122,3 +123,16 @@ def obj_or_id(parameter, param_name, object_types):
         obj_type_list = ",".join([obj_type.__name__ for obj_type in object_types])
         message = 'Parameter {} must be of type {} or int.'.format(param_name, obj_type_list)
         raise TypeError(message)
+
+
+def get_institution_url(base_url):
+    """
+    Given an API root URL, grab the protocol (http/https) and the host
+    and then return that, discarding the rest of the string.
+
+    :param base_url: The base URL of the API.
+    :type base_url: str
+    :rtype: str
+    """
+    parsed = urlparse(base_url)
+    return parsed.scheme + '://' + parsed.netloc
```

Comment:
This feels heavy-handed to me. It's a pretty huge assumption that every instance of Canvas ever will be exactly at the root of a site.

It's entirely possible that a University's Canvas instance could live at `https://fakeuniversity.edu/canvas`, or something similar. This method would (incorrectly) cut off the `/canvas` bit. Even if we assume all institutions do follow the same url pattern, there's no guarantee that every self-hosted development instance will.

I think we should instead specifically trim `/api/v1/`, `/api/v1`, etc. from the url instead of dropping the path.

Reply:
There's also a weird side effect where if you pass an invalid url to this function:

```
>>> get_institution_url('')
u'://'
>>> get_institution_url('test')
u'://'
>>> get_institution_url('google.com')
u'://'
```

Perhaps raise an exception if it's not valid?

Reply:
Agreed on using `netloc`. I've updated this to remove urlparse, now it'll just trim the end of a string containing /api/v1.